install.psm1

```diff
@@ -0,0 +1,261 @@
+﻿#
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#Requires -RunAsAdministrator
+
+function Get-Install-Directory([string]$InstallDir) {
```

Comment:
> Right now if the user provides -InstallDir "" it would most probably fail.

True, `Test-Path "" -IsValid` definitely won't allow it.

I think if we remove `<auto>`  then it's adding again one more necessary configuration to setup that can be easily avoided. I think we should not use auto values only if it's absolutely impossible. Just to keep the setup process as simple as possible.

Reply:
> I think if we remove <auto> then it's adding again one more necessary configuration to setup that can be easily avoided.

I mean use `""` or `$null` instead of auto 😉 See:

```
if (-not $InstallDir) {
    $InstallDir = "$env:LOCALAPPDATA\Programs\OpenTelemetry .NET AutoInstrumentation"
}
```

Reply:
My preference is only to lose the requirement to provide value by default. So eg defaulting to AppData dir would be also fine. Just wanted to see if anyone actually prefers quick switching between known dirs.

`<auto>` value is from dotnet install. I guess they just wanted to differentiate from missing value and auto value... so I though it might be usable here also, if you mistakenly invoke "" then nothing happens (besides error).

Reply:
> <auto> value is from dotnet install. I guess they just wanted to differentiate from missing value and auto value... so I though it might be usable here also, if you mistakenly invoke "" then nothing happens (besides error).

You convinced me to leave it now as it is 😉 We can always change it later.